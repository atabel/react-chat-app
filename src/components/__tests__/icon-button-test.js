// @flow
import * as React from 'react';
import IconButton from '../icon-button';
import renderer from 'react-test-renderer';

test('Renders an img when a url is passed', () => {
    const tree = renderer.create(<IconButton icon="some/url" label="some title" onPress={() => {}} />);

    expect(tree).toMatchSnapshot();
});

test('Renders an icon as child when a react element is passed', () => {
    const tree = renderer.create(<IconButton icon={<svg>whatever</svg>} onPress={() => {}} />);

    expect(tree).toMatchSnapshot();
});
