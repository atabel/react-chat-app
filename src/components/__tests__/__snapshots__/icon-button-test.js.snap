// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders an icon as child when a react element is passed 1`] = `
<div
  onClick={[Function]}
  style={undefined}
>
  <svg>
    whatever
  </svg>
</div>
`;

exports[`Renders an img when a url is passed 1`] = `
<img
  alt="some title"
  onClick={[Function]}
  src="some/url"
  style={undefined}
/>
`;
